File: move2alf-pipeline/src/main/scala/eu/xenit/move2alf/pipeline/actions/BasicAction.scala
package eu.xenit.move2alf.pipeline.actions

import eu.xenit.move2alf.pipeline.actors.M2AActor
import eu.xenit.move2alf.pipeline.AbstractMessage


/**
 * Created with IntelliJ IDEA.
 * User: thijs
 * Date: 3/5/13
 * Time: 2:54 PM
 * To change this template use File | Settings | File Templates.
 */
abstract class AbstractAction {

  private[actions] var actor: AbstractM2AActor = _

  final protected def setStateValue(key:String, value:Any) {
    actor.setStateValue(key, value)
  }

  final protected def getStateValue(key: String): Any = {
    actor.getStateValue(key)
  }
}

File: move2alf-pipeline/src/main/java/eu/xenit/move2alf/pipeline/actions/BasicAction.java
package eu.xenit.move2alf.pipeline.actions;

import eu.xenit.move2alf.pipeline.AbstractMessage;
import eu.xenit.move2alf.pipeline.actions.context.SendingContext;

/**
 * Created with IntelliJ IDEA.
 * User: thijs
 * Date: 3/5/13
 * Time: 4:44 PM
 * To change this template use File | Settings | File Templates.
 */
public interface BasicAction<T extends AbstractMessage, U extends AbstractMessage>{

    public void executeImpl(T message, SendingContext<U> context);

}
